### ISSUE TITLE
DOC: Validate that items in See Also section do not start by pandas.

### ISSUE THREAD

**Issue by datapythonista** [pandas-dev/pandas#23136]
`scripts/validate_docstrings.py` validates that the content of a docstring follows our standards. There are still some of those standards that the script does not validate, and this gives the users the wrong impression that the docstring is all right, when it's not really the case. All the missing validations are listed in #20298.

One of the standards is that the items in the `See Also` section should not start by `pandas.` as this is already configured in sphinx to be the default. So, instead of `pandas.DataFrame.head : Return the first n rows` we prefer `DataFrame.head : Return the first n rows`, as `pandas` is the default package,

For this issue is required:
- Change `scripts/validate_docstings.py` to give an error if the items have the prefix `pandas.`.
- Add tests in `scripts/tests/test_validate_docstrings.py`
- Run the script for all docstrings (i.e. ./scripts/validate_docstings.py) and check in the resulting report how many of the docstrings in pandas have these imports. Add as a comment in this issue the list, unless they are a reduced number, in which case they can be directly fixed in the same PR.

**Comment by thoo** [comment]
I found 425 lines which starts with prefix `pandas.`. I will submit my PR now but please let me know if you want me to fix with this PR.

**Comment by datapythonista** [comment]
I think it's better to keep them in a separate PR. Once this is fixed we can create few issues to fix them based on their files, so people can work on them in parallel, as it's quite a lot of them.
